pkg/daemon/util/copybins.go

```diff
@@ -36,6 +36,7 @@ func CopyBinaries(target string) error {
 	return copyBinary(defaultTiniDir, target, "tini")
 }
 
+// #nosec G307 Calling defer to close the file without checking the error return is not a risk for a simple file open and close
```

Comment:
It is not safe to defer closing files that are open for writing

Reply:
@richardelling thanks for your comment, I'm taking another look at this and see that the following pattern would be much better:
1. Open the file
1. `defer file.Close()`
1.  Write to the file
1. `file.Close()` --> Return an error if the close fails

This way, the file is guaranteed to close, if there is some write error, the original error will be returned, and if there is an error closing the file we will return the error. `Close()` will effectively be called twice in the normal course of execution, but the second close is a no-op.